3723/1/represent_page.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'sorted_pages'
+
+class RepresentPages
+  attr_reader :rows
+
+  def initialize(rows)
+    @rows = rows
+  end
+
+  def call(option = nil)
+    if option
+      sort_by_number(option) if option.is_a? Integer
+      sort_by_name(option) if option.is_a? String
+    end
+    represent_info
+  end
+
+  private
+
+  def sort_by_number(number)
+    @rows = @rows.first(number) if number.positive?
+  end
+
+  def sort_by_name(name)
+    @rows.map do |row|
+      if row.first.include?(name)
+        @rows = []
```

Comment:
It looks wrong. You change the object you iterate on, never do that. It won't sort as it should according by its name, `@rows` will have either unsorted elements in the end or only one last row that includes `name`. Maybe you actually do filtration? Or even looking for a single row that fits the statement?

Reply:
It's the point where the specs come. Cover the classes by tests, it will help both you to study and me to make sure of correctness.

Reply:
Done